source/common/router/config_impl.h

```diff
@@ -86,6 +88,8 @@ class VirtualHostImpl : public VirtualHost {
   const std::string& name() const override { return name_; }
   const RateLimitPolicy& rateLimitPolicy() const override { return rate_limit_policy_; }
 
+  RequestHeaderParser& requestHeaderParser() const { return *request_headers_parser_; };
```

Comment:
if i remove the const, i see the following error:

source/common/router/config_impl.cc:159:3: error: member function 'requestHeaderParser' not viable: 'this' argument has type 'const Envoy::Router::VirtualHostImpl', but function is not marked const
  vhost_.requestHeaderParser().evaluateRequestHeaders(headers, requestInfo,
  ^~~~~~